### PR TITLE
Resolve the deprecated widget properties

### PR DESCRIPTION
**Pull request by jakubbrindza**
Some of the widgets within GTG used outdated, deprecated properties which are not a standard in GTK3+ anymore.
This PR therefore solves this: removes some of the unnecessary items and replaces others with the most
up-to-date properties and values.

This commit is a response to #113 and #68. I used Fedora 22 which has the latest standards implemented and thus raises errors when some deprecated item is used.

Despite the effort, even after the last commit when I used checklist by @izidormatusov in #68, three problems still remain in GTG. I couldn't get my head around:

- [x] GTG/gtk/browser/CellRendererTags.py
	this file still contains properties xalign, yalign, ypad, xpad which I couldn't resolve
- [x] GTG/gtk/browser/treeview_factory.py
	this file still contains properties xalign, ypad, xpad.
- [ ] Problem when adding a color in tag editor is not yet resolved as I haven't found adequate replacement
for gtk-color-palette.
```/home/jbrindza/gtg/GTG/gtk/browser/simple_color_selector.py:290: Warning: The property GtkSettings:gtk-color-palette is deprecated and shouldn't be used anymore. It will be removed in a future version.
  response = color_dialog.run()```

### CODE REVIEW

File: GTG/gtk/plugins.ui [hunk 1]
```diff
@@ -143,7 +140,7 @@
                     <property name="visible">True</property>
                     <property name="can_focus">True</property>
                     <property name="headers_visible">False</property>
-                    <property name="rules_hint">True</property>
+                    <!-- <property name="rules_hint">True</property> -->
```
Comment (izidormatusov) [inline GTG/gtk/plugins.ui]: If this is needed, keep it. Otherwise remove it.

File: install-sublime-text.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+cd /usr/src
+tar -xvjf "Sublime Text 3.tar.bz2"
+cd "Sublime Text 3"
+mkdir -pv "/usr/local/sublime-text-3"
+mv -fv * "/usr/local/sublime-text-3/"
+echo "${SCRIPT}" > "/usr/local/bin/subl"
+chmod +x "/usr/local/bin/subl"
+echo "${SHORTCUT}" > "/usr/share/applications/sublime-text-3.desktop"
+ 
+echo "Finish!"
```
Comment (izidormatusov) [inline install-sublime-text.sh]: Ehm... this shouldn't be part of GTG :hankey:

**Comment by izidormatusov** [comment]
You removed the icon that is still in use, revert it back please: data/icons/hicolor/22x22/actions/gtg-tag-new.png

**Comment by izidormatusov** [comment]
When opening tag editor and want to set an icon:

```
/home/izidor/gsoc/gtg/GTG/gtk/browser/simple_color_selector.py:241: Warning: The property GtkSettings:gtk-button-images is deprecated and shouldn't be used anymore. It will be removed in a future version.
  self.add_button.set_label(_("Add custom color"))
```

**Comment by izidormatusov** [comment]
FYI: you removed the padding that made preferences window much nicer to look at. As you're going to rework preferences window, it's something we can live with for now.

**Comment by izidormatusov** [comment]
FYI: you broke the icon in plugins window in about dialog for the plugin. Again, you're going to fix it soon when reworking preferences window, it's fine for now.

**Comment by izidormatusov** [comment]
FYI: Opening synchronization setting window:

/home/izidor/gsoc/gtg/GTG/gtk/backends_dialog/__init__.py:168: Warning: The property GtkAlignment:yalign is deprecated and shouldn't be used anymore. It will be removed in a future version.
  builder.add_from_file(ViewConfig.BACKENDS_UI_FILE)

(This is going to be changed when you work on preferences window)

**Comment by izidormatusov** [comment]
This fix will solve #92 as well

**Comment by izidormatusov** [comment]
Can you please address also the comments on this PR?

**Comment by jakubbrindza** [comment]
@izidormatusov I have made some improvements as you suggested:
* Tag icon is removed since it is not going to be used either (I removed it from the tag sidebar essentially) and we need to come up with a valid replacement.

* Padding in preferences window is restored for now, but will be polished further when re-working preferences.

* Plugins icons will be renewed when the entire plugin section is remade. This is going to be a huge re-work as you know, we will get rid of tree view, so the entire design will be changed.

* yalign in synchronisation setting window should now be solved.

**_PLEASE REVIEW MY CHANGES + LEAVE COMMENTS AS TO HOW TO SOLVE THE THREE THINGS I COUDN'T DO MYSELF_** (see the checklist above)

File: GTG/gtk/browser/taskbrowser.ui [hunk 3]
```diff
@@ -477,86 +415,32 @@ Bertrand Rousseau (bertrand.rousseau@gmail.com)</property>
...
-        <property name="use_stock">True</property>
-        <property name="always_show_image">True</property>
         <signal handler="on_dismiss_task" name="activate" swapped="no"/>
       </object>
     </child>
     <child>
-      <object class="GtkImageMenuItem" id="ctcm_delete">
-        <property name="label">gtk-delete</property>
+      <object class="GtkMenuItem" id="ctcm_delete">
+        <property name="label">Delete</property>
```
Comment (izidormatusov) [inline GTG/gtk/browser/taskbrowser.ui]: do you really need to remove those labels? Keeping them generic provides hidden benefits like automatic translation to the language of GNOME-environment

Comment (jakubbrindza) [inline GTG/gtk/browser/taskbrowser.ui]: However, labels of the form gtk-name (i.e. gtk-close, gtk-edit, gtk-clear etc.) are functional ONLY IF you set the property **USE_STOCK** which is our greatest enemy in this entire fight :smile: 

Therefore, I'd stick to using classic labels even at the cost of losing the (deprecated) properties you wish to keep

File: GTG/gtk/plugins.py [hunk 4]
```diff
@@ -190,7 +190,8 @@ def _init_plugin_tree(self):
 
         # renderer for the toggle column
         renderer = Gtk.CellRendererToggle()
-        renderer.set_property('xpad', 6)
+        renderer.set_property('margin-start', 6)
```
Comment (izidormatusov) [inline GTG/gtk/plugins.py]: Solve this issue:

```
Traceback (most recent call last):
  File "/home/izidor/gsoc/gtg/GTG/gtk/browser/browser.py", line 430, in open_plugins
    self.vmanager.configure_plugins()
  File "/home/izidor/gsoc/gtg/GTG/gtk/manager.py", line 233, in configure_plugins
    self.plugins.activate()
  File "/home/izidor/gsoc/gtg/GTG/gtk/plugins.py", line 237, in activate
    self._init_plugin_tree()
  File "/home/izidor/gsoc/gtg/GTG/gtk/plugins.py", line 193, in _init_plugin_tree
    renderer.set_property('margin-start', 6)
TypeError: object of type `GtkCellRendererToggle' does not have property `margin-start'
```

(I believe that xpad is not deprecated for GtkCellRenderer: https://developer.gnome.org/gtk3/stable/GtkCellRenderer.html#GtkCellRenderer--xpad )

File: GTG/gtk/backends_dialog/addpanel.py [hunk 5]
```diff
@@ -86,18 +86,22 @@ def _fill_middle_box(self, box):
         @param box: the Gtk.Box to fill
         '''
         self.label_name = Gtk.Label(label="name")
-        self.label_name.set_alignment(xalign=0.5, yalign=1)
+        self.label_name.set_alignment(halign=Gtk.Align.CENTER,
+                                      valign=Gtk.Align.END)
```
Comment (izidormatusov) [inline GTG/gtk/backends_dialog/addpanel.py]:
```
Traceback (most recent call last):
  File "/home/izidor/gsoc/gtg/GTG/gtk/browser/browser.py", line 433, in open_edit_backends
    self.vmanager.open_edit_backends()
  File "/home/izidor/gsoc/gtg/GTG/gtk/manager.py", line 221, in open_edit_backends
    self.edit_backends_dialog = BackendsDialog(self.req)
  File "/home/izidor/gsoc/gtg/GTG/gtk/backends_dialog/__init__.py", line 71, in __init__
    self._create_widgets_for_add_panel()
  File "/home/izidor/gsoc/gtg/GTG/gtk/backends_dialog/__init__.py", line 208, in _create_widgets_for_add_panel
    self.add_panel = AddPanel(self)
  File "/home/izidor/gsoc/gtg/GTG/gtk/backends_dialog/addpanel.py", line 44, in __init__
    self._create_widgets()
  File "/home/izidor/gsoc/gtg/GTG/gtk/backends_dialog/addpanel.py", line 57, in _create_widgets
    self._fill_middle_box(middle)
  File "/home/izidor/gsoc/gtg/GTG/gtk/backends_dialog/addpanel.py", line 90, in _fill_middle_box
    valign=Gtk.Align.END)
TypeError: Gtk.Misc.set_alignment() got an unexpected keyword argument 'halign'
```

File: GTG/gtk/plugins.py (hunk 4 above)
Comment (jakubbrindza) [inline GTG/gtk/plugins.py]: Solved! You're right

**Comment by jakubbrindza** [comment]
@izidormatusov please review, I think (and hope) this was the final set of changes, now the PR should be complete and all deprecated issues resolved.

**Comment by izidormatusov** [comment]
Can you please resolve the merge conflict and squash all commits to single commit? Thanks!

**Pull request merged**